### PR TITLE
Add Metrics

### PR DESCRIPTION
**Pull request by pdewilde**
Adds metrics from https://github.com/abcxyz/abc-updater

See updates to README.md for more information.

### CODE REVIEW

File: cmd/abc/abc.go
Comment (drevell) [inline cmd/abc/abc.go:143]: I know this was already like this, but we can invert this conditional and save one level of nesting.

```
if version.Version == "source" {
    return func(){}
```

File: cmd/abc/abc.go
Comment (drevell) [inline cmd/abc/abc.go:34]: The name "wrapper" could be more helpful; all it says is that something is being wrapped but not what it is

**Review by drevell — CHANGES_REQUESTED** [review]

File: cmd/abc/abc.go
Comment (pdewilde) [reply cmd/abc/abc.go:170]: Honestly, we could swap, I just figured I would start timing execution from as early as possible, though checkSupportedOS() won't meaningfully change the runtime

File: cmd/abc/abc.go
Comment (pdewilde) [reply cmd/abc/abc.go:34]: Wasn't sure what to call it, since "metrics" would be a bit confusing (would need to alias one of the imports anyways).

Figured other wrappers could be put in the same package. Open to naming suggestions.

File: cmd/abc/abc.go
Comment (pdewilde) [reply cmd/abc/abc.go]: I figured failing to check for updates isn't super actionable to most users.

Imagine a future where the update server is no longer running, a user probably doesn't want a "failed to check for updates" message on every single invocation.

File: cmd/abc/abc.go
Comment (pdewilde) [reply cmd/abc/abc.go]: Oops, was just when I was doing an experiment earlier but forgot to fix when I went back and made everything high quality

File: cmd/abc/abc.go
Comment (pdewilde) [reply cmd/abc/abc.go]: Good call

File: cmd/abc/abc.go
Comment (pdewilde) [reply cmd/abc/abc.go]: moved

File: cmd/abc/abc.go
Comment (pdewilde) [reply cmd/abc/abc.go]: Added log

File: cmd/abc/abc.go
Comment (pdewilde) [reply cmd/abc/abc.go]: Done, though I moved the default timeout into the wrapper as it made more sense there.

File: cmd/abc/abc.go
Comment (drevell) [reply cmd/abc/abc.go:170]: Ah, I see, sounds good.

File: cmd/abc/abc.go
Comment (drevell) [reply cmd/abc/abc.go:34]: How about "metricswrap"? Let's plan on putting each future wrapper in its own package; it's nice to have meaningful package names.

**Review by drevell — APPROVED** [review]

**Comment by drevell** [comment]
The PR description says "DO NOT MERGE", is that intentional? I'm reviewing this as if its intended to be merged eventually.

**Comment by pdewilde** [comment]
> The PR description says "DO NOT MERGE", is that intentional? I'm reviewing this as if its intended to be merged eventually.

Yes, waiting on feedback on privacy policy language before I merge.

**Pull request merged**
